models/events/build.go

```diff
@@ -0,0 +1,60 @@
+package events
+
+import (
+	"time"
+
+	"github.com/gofrs/uuid"
+)
+
+type EventBuilder struct {
+	event Event
+}
+
+func NewEvent() *EventBuilder {
+	operationId, _ := uuid.NewV4()
+	return &EventBuilder{
+		event: Event{
+			CreatedAt:   time.Now(),
+			OperationID: operationId,
+		},
+	}
+}
+
+func (e *EventBuilder) ActedOn(resource uuid.UUID) *EventBuilder {
+	e.event.ActedUpon = resource
+	return e
+}
+
+func (e *EventBuilder) Description(description string) *EventBuilder {
+	e.event.Description = description
+	return e
+}
+
+func (e *EventBuilder) WithEventType(eventType string) *EventBuilder {
+	e.event.EventType = eventType
+	return e
+}
+
+func (e *EventBuilder) AddMetadata(metadata interface{}) *EventBuilder {
+	e.event.Metadata = metadata
+	return e
+}
+
+func (e *EventBuilder) WithSeverity(severity EventSeverity) *EventBuilder {
+	e.event.Severity = severity
+	return e
+}
+
+func (e *EventBuilder) InitiatedBy(id uuid.UUID) *EventBuilder {
```

Comment:
@MUzairS15 follow the table design spec exactly. Do not rename attributes.

Reply:
@leecalcote I have not diverged from the schema, the attribute name are exactly the same https://github.com/meshery/meshkit/pull/356/files/83dcd0952d500438cda2161e305b8ed1d12547a8#diff-5bc079c95a1811fd042d036db5a54fbae0ba3ef9e1645ce28d4eb4050b749049R49

Reply:
Oh, I see. I'm confusing attribute name and function name. Symmetric function naming could be `fromUser` or InitiatingUser and InitiatingSystem.